tsdb/engine/tsm1/encoding.gen.go.tmpl

```diff
@@ -53,18 +53,23 @@ func (a {{.Name}}Values) assertOrdered() {
 // Deduplicate returns a new slice with any values that have the same timestamp removed.
 // The Value that appears last in the slice is the one that is kept.
 func (a {{.Name}}Values) Deduplicate() {{.Name}}Values {
-	m := make(map[int64]{{.Name}}Value)
-	for _, val := range a {
-		m[val.UnixNano()] = val
+	if len(a) == 0 {
+		return a
 	}
+	sort.Stable(a)
```

Comment:
Yes.  It's important that it's a stable sort because duplicates (t1 == t2), may have different values and we need the values added later to overwrite existing values.  If it's not a stable sort, the ordering could get swapped causing the latest write to be lost.